CMakeLists.txt

```diff
@@ -119,6 +125,19 @@ set_target_properties(Spglib_symspg PROPERTIES
 		EXPORT_NAME symspg
 		OUTPUT_NAME symspg)
 add_library(Spglib::symspg ALIAS Spglib_symspg)
+
+# Write spglib_export.h to the current binary directory
+include(GenerateExportHeader)
+generate_export_header(Spglib_symspg
+	EXPORT_FILE_NAME spglib_export.h
+	EXPORT_MACRO_NAME API_SPGLIB
+)
```

Comment:
Can you post a copy of the content of the generated `spglib_export.h` file?

I have a few design questions about this:
- Are the macros compiler independent, specifically on linux, would it be applicable for all compilers?
- On windows, how do `gcc` vs `msvc` compilers work when they are cross-compiler consumed?
- For Unix I would prefer to minimize the files populated in `/usr/include`, and I'm not sure if it merits populating a `spglib` folder for this. Probably it would work if the 2 files are concatenated before install, have you seen such design implemented?

Reply:
```c
#ifndef API_SPGLIB_H
#define API_SPGLIB_H

#ifdef SPGLIB_SYMSPG_STATIC_DEFINE
#  define API_SPGLIB
#  define SPGLIB_SYMSPG_NO_EXPORT
#else
#  ifndef API_SPGLIB
#    ifdef Spglib_symspg_EXPORTS
        /* We are building this library */
#      define API_SPGLIB __attribute__((visibility("default")))
#    else
        /* We are using this library */
#      define API_SPGLIB __attribute__((visibility("default")))
#    endif
#  endif

#  ifndef SPGLIB_SYMSPG_NO_EXPORT
#    define SPGLIB_SYMSPG_NO_EXPORT __attribute__((visibility("hidden")))
#  endif
#endif

#ifndef SPGLIB_SYMSPG_DEPRECATED
#  define SPGLIB_SYMSPG_DEPRECATED __attribute__ ((__deprecated__))
#endif

#ifndef SPGLIB_SYMSPG_DEPRECATED_EXPORT
#  define SPGLIB_SYMSPG_DEPRECATED_EXPORT API_SPGLIB SPGLIB_SYMSPG_DEPRECATED
#endif

#ifndef SPGLIB_SYMSPG_DEPRECATED_NO_EXPORT
#  define SPGLIB_SYMSPG_DEPRECATED_NO_EXPORT SPGLIB_SYMSPG_NO_EXPORT SPGLIB_SYMSPG_DEPRECATED
#endif

#if 0 /* DEFINE_NO_DEPRECATED */
#  ifndef SPGLIB_SYMSPG_NO_DEPRECATED
#    define SPGLIB_SYMSPG_NO_DEPRECATED
#  endif
#endif

#endif /* API_SPGLIB_H */
```

Reply:
This is the GCC and clang version for UNIX.
On Windows with DLL the  export/import statement are different!

The 2 files seems ok for me in /usr/include, but a namespace subdirectory would break your API.

Reply:
The windows one is the one I want to see, as well as a gcc version on windows. I need to think if it makes sense to have it in a compiler specific version and what would happen with cross-compilation. There is conda and julia to consider.

Reply:
```c
#ifndef API_SPGLIB_H
#define API_SPGLIB_H

#ifdef SPGLIB_SYMSPG_STATIC_DEFINE
#  define API_SPGLIB
#  define SPGLIB_SYMSPG_NO_EXPORT
#else
#  ifndef API_SPGLIB
#    ifdef Spglib_symspg_EXPORTS
        /* We are building this library */
#      define API_SPGLIB __declspec(dllexport)
#    else
        /* We are using this library */
#      define API_SPGLIB __declspec(dllimport)
#    endif
#  endif

#  ifndef SPGLIB_SYMSPG_NO_EXPORT
#    define SPGLIB_SYMSPG_NO_EXPORT 
#  endif
#endif

#ifndef SPGLIB_SYMSPG_DEPRECATED
#  define SPGLIB_SYMSPG_DEPRECATED __declspec(deprecated)
#endif

#ifndef SPGLIB_SYMSPG_DEPRECATED_EXPORT
#  define SPGLIB_SYMSPG_DEPRECATED_EXPORT API_SPGLIB SPGLIB_SYMSPG_DEPRECATED
#endif

#ifndef SPGLIB_SYMSPG_DEPRECATED_NO_EXPORT
#  define SPGLIB_SYMSPG_DEPRECATED_NO_EXPORT SPGLIB_SYMSPG_NO_EXPORT SPGLIB_SYMSPG_DEPRECATED
#endif

#if 0 /* DEFINE_NO_DEPRECATED */
#  ifndef SPGLIB_SYMSPG_NO_DEPRECATED
#    define SPGLIB_SYMSPG_NO_DEPRECATED
#  endif
#endif

#endif /* API_SPGLIB_H */
```